### ISSUE TITLE
error: cannot pull with rebase: You have unstaged changes.

### ISSUE THREAD

**Issue by tienviitta** [ohmybash/oh-my-bash#407]
This showed up today when trying to update oh-my-bash:

❯ upgrade_oh_my_bash
Updating Oh My Bash
error: cannot pull with rebase: You have unstaged changes.
error: please commit or stash them.
oh-my-bash: running 'git rebase --abort'...
fatal: No rebase in progress?
There was an error updating.
If you have uncommited changes in '/home/peva/.oh-my-bash', please commit, stash or discard them and retry updating.
If you have your own local commits in '/home/peva/.oh-my-bash' that conflict with the upstream changes, please resolve conflicts and merge the upstream manually.
 
I don't recall doing anything at .oh-my-bash.

**Comment by akinomyoga** [comment]
But it says there are unstaged changes. I don't think Git will tell a lie. In some systems, the operating system can change the file attributes on some actions even if you don't directly touch the files. To check what files have been changed, you can run the following commands:

```bash
$ cd ~/.oh-my-bash
$ git status
```

To check the changes, you can run the following commands:

```bash
$ cd ~/.oh-my-bash
$ git diff
```

If `git diff` opened `less` (a text viewer in the terminal), you can quit it by typing the key `q`.

**Comment by tienviitta** [comment]
Thanks, it seems that some Ubuntu 22.04 (?) update did this:
diff --git a/templates/bash_profile.osh-template b/templates/bash_profile.osh-template
deleted file mode 100644
index 7872597..0000000
--- a/templates/bash_profile.osh-template
+++ /dev/null
@@ -1,11 +0,0 @@
-# .bash_profile -*- mode: sh -*-
-
-# Load login settings and environment variables
-if [[ -f ~/.profile ]]; then
-  source ~/.profile
-fi
-
-# Load interactive settings
-if [[ -f ~/.bashrc ]]; then
-  source ~/.bashrc
-fi

**Comment by akinomyoga** [comment]
> Thanks, it seems that some Ubuntu 22.04 (?) update did this:

OK, diff tells that the file `templates/bash_profile.osh-template` is removed or moved to another place. I don't think the Ubuntu update is related. Maybe you have moved the file to `~/.bashrc` or somewhere. Or there was some problem in the storage (HDD SSD, etc.) or the filesystem (e.g. by a sudden shutdown). If you are really sure that you didn't anything, it might be a signal that your HDD/SSD will soon be broken, so you might want to be careful backing up your files, etc.

If you want to fix the current situation, you can check out the removed file with the following command:

```bash
$ cd ~/.oh-my-bash
$ git checkout @ templates/bash_profile.osh-template
```

After that you should be able to upgrade it by `upgrade_oh_my_bash`.

**Comment by akinomyoga** [comment]
Ah, I'm sorry. I realized a serious problem in `install.sh`. This is a problem applied after https://github.com/ohmybash/oh-my-bash/issues/302#issuecomment-1416781059. I'll fix it. Thank you for your report.

**Comment by akinomyoga** [comment]
fixed ebd541c.
